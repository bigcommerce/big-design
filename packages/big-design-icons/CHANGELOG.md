# Change Log

## 1.2.2

### Patch Changes

- 98fbd46: feat(component): Add an action prop for columns whose type is number or text

## 1.2.1

### Patch Changes

- 69e7772: Remove Pin icon hardcoded fill colour

## 1.2.0

### Minor Changes

- 0f8cf34: Added Pin icon

## 1.1.0

### Minor Changes

- 01a5b20: Added AutoAwesome icon from Material Icons library

## 1.0.3

### Patch Changes

- Updated dependencies [f8665f3]
  - @bigcommerce/big-design-theme@1.1.0

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.2](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@1.0.1...@bigcommerce/big-design-icons@1.0.2) (2024-08-06)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [1.0.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@1.0.0...@bigcommerce/big-design-icons@1.0.1) (2024-06-11)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [1.0.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.26.1...@bigcommerce/big-design-icons@1.0.0) (2024-04-17)

### Features

- export types without reassignment ([9bf2244](https://github.com/bigcommerce/big-design/commit/9bf224417433e54680794f97d30a01fb77b24d69))

## [0.26.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.26.0...@bigcommerce/big-design-icons@0.26.1) (2024-04-09)

### Bug Fixes

- typescript build output with wrong imports ([a812497](https://github.com/bigcommerce/big-design/commit/a81249798e20c10754c938b2b3d3dcc95d81aa0c))

# [0.26.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.25.1...@bigcommerce/big-design-icons@0.26.0) (2024-04-02)

### chore

- **deps-dev:** bump flag-icons from 6.11.0 to 7.1.0 ([#1345](https://github.com/bigcommerce/big-design/issues/1345)) ([54572b6](https://github.com/bigcommerce/big-design/commit/54572b64e54cb5bb33d28c41a5cfe74844040d40))

### BREAKING CHANGES

- **deps-dev:** This upgrade removes a few flag icons. See the commit diff to check which
  ones were removed and ensure you are not using them in your application code.

## [0.25.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.25.0...@bigcommerce/big-design-icons@0.25.1) (2024-01-23)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.25.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.24.0...@bigcommerce/big-design-icons@0.25.0) (2023-10-09)

### Bug Fixes

- **icons:** downgrade 'camelcase' to 6.3.0 ([#1284](https://github.com/bigcommerce/big-design/issues/1284)) ([1004097](https://github.com/bigcommerce/big-design/commit/100409789ab0beb9095eea3580aed7b5b3fc8776))

### Features

- **icons:** add Draft icon ([#1302](https://github.com/bigcommerce/big-design/issues/1302)) ([3ae2423](https://github.com/bigcommerce/big-design/commit/3ae2423ccaf2b53f282790218dcb80d622954241))
- **icons:** add FileDownloadIcon ([#1286](https://github.com/bigcommerce/big-design/issues/1286)) ([7d73767](https://github.com/bigcommerce/big-design/commit/7d737670533c45ab09a66556b1d4b2316331ff03))
- **icons:** add window expand and minimize icons ([997400b](https://github.com/bigcommerce/big-design/commit/997400b4ba5dc8d4c484cb690ee335a112de6c32))

# [0.24.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.23.2...@bigcommerce/big-design-icons@0.24.0) (2023-07-26)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.23.2](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.23.1...@bigcommerce/big-design-icons@0.23.2) (2023-06-14)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.23.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.23.0...@bigcommerce/big-design-icons@0.23.1) (2023-06-09)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.23.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.22.0...@bigcommerce/big-design-icons@0.23.0) (2023-04-05)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.22.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.21.0...@bigcommerce/big-design-icons@0.22.0) (2023-01-09)

### Features

- convert to react@18 only ([#1044](https://github.com/bigcommerce/big-design/issues/1044)) ([06c4697](https://github.com/bigcommerce/big-design/commit/06c469721bc06d1f872bed6bf5a46e6b568644e5))

### BREAKING CHANGES

- Requires `react@18` or higher.

# [0.21.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.20.1...@bigcommerce/big-design-icons@0.21.0) (2022-09-27)

### Features

- **deps:** react@18 support ([51c688b](https://github.com/bigcommerce/big-design/commit/51c688b282277964eb01c1ed67c5c27cc0bc4c3e))

## [0.20.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.20.0...@bigcommerce/big-design-icons@0.20.1) (2022-09-27)

### Features

- **deps:** unfork from flag-icons dependency ([ac723e3](https://github.com/bigcommerce/big-design/commit/ac723e384ec3c482d78107558cd31e7793a515f1))

# [0.20.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.19.2-alpha.0...@bigcommerce/big-design-icons@0.20.0) (2022-08-17)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.19.2-alpha.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.19.1...@bigcommerce/big-design-icons@0.19.2-alpha.0) (2022-08-10)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.19.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.19.0...@bigcommerce/big-design-icons@0.19.1) (2022-07-12)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.19.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.18.1...@bigcommerce/big-design-icons@0.19.0) (2022-07-06)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.18.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.18.0...@bigcommerce/big-design-icons@0.18.1) (2022-06-10)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.18.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.2-alpha.1...@bigcommerce/big-design-icons@0.18.0) (2022-05-11)

### Bug Fixes

- tiny-async-pool api changes ([b4c04ba](https://github.com/bigcommerce/big-design/commit/b4c04bac3556b084ac1d39d2f0fde57efd7d8f43))

## [0.17.2-alpha.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.2-alpha.0...@bigcommerce/big-design-icons@0.17.2-alpha.1) (2022-03-22)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.17.2-alpha.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.1...@bigcommerce/big-design-icons@0.17.2-alpha.0) (2021-10-28)

### Bug Fixes

- **icons:** add `aria-hidden=true` if an icon doesn't have a title ([e3bb896](https://github.com/bigcommerce/big-design/commit/e3bb896fed2d7b1afb30f0134f161cfae5a00398))
- **icons:** Ensure aira-hidden is correctly set in flags ([9f49a2d](https://github.com/bigcommerce/big-design/commit/9f49a2dfd19734e797945483cb99efe8ea1503f1))

## [0.17.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.0...@bigcommerce/big-design-icons@0.17.1) (2021-09-03)

### Features

- **icons:** Add Home/KeyboardDoubleArrowLeftIcon/SwapHoriz icons ([#583](https://github.com/bigcommerce/big-design/issues/583)) ([519655f](https://github.com/bigcommerce/big-design/commit/519655fe8ee3d619d1b85937e86635332fb77301))

# [0.17.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.0-alpha.2...@bigcommerce/big-design-icons@0.17.0) (2021-08-04)

### Features

- **icons:** add content copy icon ([#572](https://github.com/bigcommerce/big-design/issues/572)) ([4ca0b76](https://github.com/bigcommerce/big-design/commit/4ca0b76d12cc5a9d2e1dd0b90d90bc2cfb6a1294))

# [0.17.0-alpha.2](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.0-alpha.1...@bigcommerce/big-design-icons@0.17.0-alpha.2) (2021-08-02)

### chore

- **deps:** upgrade to styled-components v5 ([#566](https://github.com/bigcommerce/big-design/issues/566)) ([dd83711](https://github.com/bigcommerce/big-design/commit/dd83711797eb8aaa7a8406eebf1383116bff2420))

### BREAKING CHANGES

- **deps:** You will need to update to styled-components v5. In addition,
  you will need to import the base fonts in your <head> element.
  See the "Getting Started" page or README.md for an example.

# [0.17.0-alpha.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.17.0-alpha.0...@bigcommerce/big-design-icons@0.17.0-alpha.1) (2021-07-06)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.17.0-alpha.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.16.1...@bigcommerce/big-design-icons@0.17.0-alpha.0) (2021-06-10)

**Note:** Version bump only for package @bigcommerce/big-design-icons

## [0.16.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.16.0...@bigcommerce/big-design-icons@0.16.1) (2021-03-29)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.16.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.15.0...@bigcommerce/big-design-icons@0.16.0) (2021-03-08)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.15.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.14.1...@bigcommerce/big-design-icons@0.15.0) (2021-02-11)

### Bug Fixes

- **docs:** Specify styled-components version in the README ([#490](https://github.com/bigcommerce/big-design/issues/490)) ([c92403e](https://github.com/bigcommerce/big-design/commit/c92403edf8e29bf1afd3c9c163c37be6ffe17a6c))

### Features

- **icons:** add cloud upload icon ([#501](https://github.com/bigcommerce/big-design/issues/501)) ([aeb9aee](https://github.com/bigcommerce/big-design/commit/aeb9aeeb18a93064c61497fc439e7e82957c1a2e))
- **icons:** add flag icons ([#493](https://github.com/bigcommerce/big-design/issues/493)) ([0519c8a](https://github.com/bigcommerce/big-design/commit/0519c8aa131461e52ffe40f40b1e45771f749c0f))

## [0.14.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.14.0...@bigcommerce/big-design-icons@0.14.1) (2020-12-21)

### Features

- **icons:** add insert_drive_file icon ([#486](https://github.com/bigcommerce/big-design/issues/486)) ([b07d9c6](https://github.com/bigcommerce/big-design/commit/b07d9c6f6416854212eb447d216117a40edb6b8e))

# [0.14.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.13.1...@bigcommerce/big-design-icons@0.14.0) (2020-11-12)

### Bug Fixes

- webpack 5 compatibility ([#466](https://github.com/bigcommerce/big-design/issues/466)) ([7e66238](https://github.com/bigcommerce/big-design/commit/7e66238ca42cb27d91b1a80cc9e1c8014808e27b))

## [0.13.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.13.0...@bigcommerce/big-design-icons@0.13.1) (2020-08-31)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.13.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.12.0...@bigcommerce/big-design-icons@0.13.0) (2020-08-13)

### Features

- **icons:** add code, brush and get_app icons ([#433](https://github.com/bigcommerce/big-design/issues/433)) ([bb48f2d](https://github.com/bigcommerce/big-design/commit/bb48f2d6aefa627f35137f6043b582268ac76ef4))

# [0.12.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.11.1...@bigcommerce/big-design-icons@0.12.0) (2020-06-16)

### Features

- **component:** add Tree component ([#406](https://github.com/bigcommerce/big-design/issues/406)) ([c67643c](https://github.com/bigcommerce/big-design/commit/c67643cfdf425d078a1ea09bce12575e35f442e7))

## [0.11.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.11.0...@bigcommerce/big-design-icons@0.11.1) (2020-04-30)

### Bug Fixes

- type regression for ts < 3.8 ([#395](https://github.com/bigcommerce/big-design/issues/395)) ([f476f9b](https://github.com/bigcommerce/big-design/commit/f476f9b3c2a950bb3bd3353a4ce180e994e465c0))

# [0.11.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.10.0...@bigcommerce/big-design-icons@0.11.0) (2020-04-29)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.10.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.9.0...@bigcommerce/big-design-icons@0.10.0) (2020-04-20)

### Bug Fixes

- **icons:** contained aria-labelledby when title prop was undefined ([#374](https://github.com/bigcommerce/big-design/issues/374)) ([f0d4df7](https://github.com/bigcommerce/big-design/commit/f0d4df70894b364ec932dfc5c62cebda5dd10bda))

# [0.9.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.8.0...@bigcommerce/big-design-icons@0.9.0) (2020-03-25)

**Note:** Version bump only for package @bigcommerce/big-design-icons

# [0.8.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.7.0...@bigcommerce/big-design-icons@0.8.0) (2020-02-21)

### Features

- **component:** add Alerts, InlineAlerts, and Message components ([#340](https://github.com/bigcommerce/big-design/issues/340)) ([d541276](https://github.com/bigcommerce/big-design/commit/d54127603fba47b46cb35c3db4caf53ab24bafc3))

# [0.7.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.6.0...@bigcommerce/big-design-icons@0.7.0) (2020-02-06)

### Bug Fixes

- add react-hooks eslint and fix missing deps/issues ([#330](https://github.com/bigcommerce/big-design/issues/330)) ([da3fbd6](https://github.com/bigcommerce/big-design/commit/da3fbd68181e98e43a95de7fce9956be91afc9b8))

### Features

- **icons:** forwardRef to svg ([#312](https://github.com/bigcommerce/big-design/issues/312)) ([0088444](https://github.com/bigcommerce/big-design/commit/0088444015df4f6538d97e657f800f5718c2706e))

# [0.6.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.5.2...@bigcommerce/big-design-icons@0.6.0) (2019-12-09)

### Bug Fixes

- **icons:** only add title when available ([b2cf7f7](https://github.com/bigcommerce/big-design/commit/b2cf7f7a00a0d158c01485ee6978c2e13404c048))

### Features

- **component:** add tab trap for modals and dialogs ([dcacbf9](https://github.com/bigcommerce/big-design/commit/dcacbf96a38bef1134e2a8dcbd986f6362e0e2b7))
- **component:** ignore theme prop overrides ([24b92f9](https://github.com/bigcommerce/big-design/commit/24b92f9873fe4f51975b7bd00300c8ce73484ea3))

### BREAKING CHANGES

- **component:** `theme` prop override is no longer supported.

## [0.5.2](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.5.1...@bigcommerce/big-design-icons@0.5.2) (2019-11-14)

### Bug Fixes

- **icons:** show proper displayName for icons ([#268](https://github.com/bigcommerce/big-design/issues/268)) ([539a02a](https://github.com/bigcommerce/big-design/commit/539a02a))

## [0.5.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.5.0...@bigcommerce/big-design-icons@0.5.1) (2019-11-12)

### Features

- **all:** better tree shaking ([c0998a7](https://github.com/bigcommerce/big-design/commit/c0998a7))

# [0.5.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.4.0...@bigcommerce/big-design-icons@0.5.0) (2019-11-11)

### Features

- **all:** bump styled-components peerDependency version ([fd89fa3](https://github.com/bigcommerce/big-design/commit/fd89fa3))

### BREAKING CHANGES

- **all:** bumped peer dependency of styled-components to `^4.3.0`

# [0.4.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.3.0...@bigcommerce/big-design-icons@0.4.0) (2019-10-29)

### Features

- **icons:** add ArrowDownward and ArrowUpward icons ([c34203f](https://github.com/bigcommerce/big-design/commit/c34203f))

# [0.3.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.2.1...@bigcommerce/big-design-icons@0.3.0) (2019-09-23)

### Features

- **component:** add indeterminate state to checkboxes ([#197](https://github.com/bigcommerce/big-design/issues/197)) ([5146fdb](https://github.com/bigcommerce/big-design/commit/5146fdb))

## [0.2.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.2.0...@bigcommerce/big-design-icons@0.2.1) (2019-09-17)

### Bug Fixes

- **icons:** use new api on icon downloader ([#195](https://github.com/bigcommerce/big-design/issues/195)) ([6bae4ae](https://github.com/bigcommerce/big-design/commit/6bae4ae))

# [0.2.0](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.1.2...@bigcommerce/big-design-icons@0.2.0) (2019-08-20)

### Features

- **icons:** add BaselineHelpIcon ([#180](https://github.com/bigcommerce/big-design/issues/180)) ([e2387cf](https://github.com/bigcommerce/big-design/commit/e2387cf))

## [0.1.2](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.1.1...@bigcommerce/big-design-icons@0.1.2) (2019-08-19)

### Bug Fixes

- **icons:** now treeshakeable ([#178](https://github.com/bigcommerce/big-design/issues/178)) ([fb3773d](https://github.com/bigcommerce/big-design/commit/fb3773d))

## [0.1.1](https://github.com/bigcommerce/big-design/compare/@bigcommerce/big-design-icons@0.1.0...@bigcommerce/big-design-icons@0.1.1) (2019-08-15)

### Features

- **docs:** add responsive mobile header ([#165](https://github.com/bigcommerce/big-design/issues/165)) ([dca031a](https://github.com/bigcommerce/big-design/commit/dca031a))

# 0.1.0 (2019-08-13)

### Bug Fixes

- **icons:** point to the correct type definitions ([f6ffb48](https://github.com/bigcommerce/big-design/commit/f6ffb48))
- **icons:** star icon now displays a star and not a square ([331e71b](https://github.com/bigcommerce/big-design/commit/331e71b))
- **icons:** use object style notation for icon base ([41a2638](https://github.com/bigcommerce/big-design/commit/41a2638))
- **theme:** issue with font sizes in createTheme ([#135](https://github.com/bigcommerce/big-design/issues/135)) ([313ff46](https://github.com/bigcommerce/big-design/commit/313ff46))

### Features

- **icons:** add color and size props ([ba07b38](https://github.com/bigcommerce/big-design/commit/ba07b38))
- **icons:** add initial batch of icons ([243d1a2](https://github.com/bigcommerce/big-design/commit/243d1a2))
- **icons:** add Notifications and FileCopy icons ([#103](https://github.com/bigcommerce/big-design/issues/103)) ([e385b9b](https://github.com/bigcommerce/big-design/commit/e385b9b))
- **icons:** initial package setup ([#89](https://github.com/bigcommerce/big-design/issues/89)) ([acfe8fe](https://github.com/bigcommerce/big-design/commit/acfe8fe))
- **icons:** use Icon postfix ([#97](https://github.com/bigcommerce/big-design/issues/97)) ([03772fb](https://github.com/bigcommerce/big-design/commit/03772fb))
