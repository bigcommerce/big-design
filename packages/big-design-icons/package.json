{
  "name": "@bigcommerce/big-design-icons",
  "version": "1.2.2",
  "sideEffects": false,
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "typings": "dist/index.d.ts",
  "license": "(MIT AND CC-BY-3.0)",
  "author": "BigCommerce Inc",
  "repository": {
    "type": "git",
    "url": "https://github.com/bigcommerce/big-design.git",
    "directory": "packages/big-design-icons"
  },
  "scripts": {
    "build": "pnpm run build:cjs && pnpm run build:es && pnpm run build:dt",
    "build:cjs": "NODE_ENV=production BABEL_ENV=cjs babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/cjs",
    "build:es": "NODE_ENV=production BABEL_ENV=es babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/es",
    "build:dt": "tsc --emitDeclarationOnly",
    "build:icons": "node scripts/build.js",
    "build:flags": "node scripts/build-flags.js",
    "download": "node scripts/downloader.js",
    "lint": "eslint . --max-warnings 0",
    "typecheck": "tsc --noEmit"
  },
  "files": [
    "dist",
    "flags",
    "svgs"
  ],
  "publishConfig": {
    "access": "public"
  },
  "prettier": "@bigcommerce/eslint-config/prettier",
  "dependencies": {
    "@babel/runtime": "^7.26.0",
    "@bigcommerce/big-design-theme": "workspace:^"
  },
  "peerDependencies": {
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "styled-components": "^5.3.5"
  },
  "devDependencies": {
    "@babel/cli": "^7.25.7",
    "@babel/core": "^7.26.0",
    "@babel/plugin-transform-runtime": "^7.25.9",
    "@babel/preset-env": "^7.26.0",
    "@babel/preset-react": "^7.26.3",
    "@babel/preset-typescript": "^7.26.0",
    "@bigcommerce/configs": "workspace:^",
    "@styled/typescript-styled-plugin": "^1.0.1",
    "@svgr/core": "^8.1.0",
    "@svgr/plugin-jsx": "^8.1.0",
    "@svgr/plugin-prettier": "^8.1.0",
    "@svgr/plugin-svgo": "^8.1.0",
    "@types/react": "^18.2.73",
    "@types/react-dom": "^18.2.23",
    "@types/styled-components": "^5.1.34",
    "babel-plugin-styled-components": "^2.0.7",
    "camelcase": "^6.3.0",
    "flag-icons": "^7.3.2",
    "fs-extra": "^11.1.0",
    "glob": "^11.0.1",
    "inquirer": "^8.2.6",
    "inquirer-autocomplete-prompt": "^2.0.1",
    "prettier": "^2.4.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-is": "^18.2.0",
    "rimraf": "^6.0.1",
    "styled-components": "^5.3.11",
    "tiny-async-pool": "^2.0.1",
    "typescript": "^5.6.3"
  }
}
