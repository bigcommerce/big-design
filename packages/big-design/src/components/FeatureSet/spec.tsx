import { AutoAwesomeIcon } from '@bigcommerce/big-design-icons';
import { render, screen } from '@testing-library/react';
import React from 'react';

import 'jest-styled-components';

import { FeatureSet } from './index';

test('render feature set', () => {
  render(<FeatureSet tags={[{ label: 'Feature 1' }, { label: 'Feature 2' }]} />);

  expect(screen.getByRole('list')).toMatchSnapshot();
});

test("doesn't forward styles", () => {
  render(
    <FeatureSet
      className="test"
      style={{ background: 'red' }}
      tags={[{ label: 'Feature 1' }, { label: 'Feature 2' }]}
    />,
  );

  expect(screen.queryByText('test')).not.toBeInTheDocument();
  expect(screen.getByRole('list')).not.toHaveStyle('background: red');
});

test("doesn't render if tags are empty", () => {
  render(<FeatureSet tags={[]} />);

  expect(screen.queryByRole('list')).not.toBeInTheDocument();
});

test('allows margins to be set', () => {
  render(<FeatureSet margin="medium" tags={[{ label: 'Feature 1' }, { label: 'Feature 2' }]} />);

  expect(screen.getByRole('list')).toHaveStyle('margin: 1rem');
});

test('renders tag with icon', () => {
  render(
    <FeatureSet
      tags={[{ label: 'Feature 1', icon: <AutoAwesomeIcon /> }, { label: 'Feature 2' }]}
    />,
  );

  const listItem = screen.getByRole('listitem', { name: 'Feature 1' });

  // eslint-disable-next-line testing-library/no-node-access
  expect(listItem.querySelector('svg')).toBeInTheDocument();
});

test("doesn't render tag with invalid label", () => {
  render(<FeatureSet tags={[{ label: '' }]} />);

  expect(screen.queryByRole('listitem')).not.toBeInTheDocument();
});
