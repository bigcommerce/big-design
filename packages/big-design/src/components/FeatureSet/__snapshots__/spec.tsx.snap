// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`render feature set 1`] = `
.c0 {
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 0.75rem;
}

.c2 {
  color: currentColor;
  margin: 0 0 1rem;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  word-wrap: normal;
  color: currentColor;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
  margin: 0;
}

.c2:last-child {
  margin-bottom: 0;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ECEEF5;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #5E637A;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  gap: 0.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-block: 0.125rem;
  padding-inline: 0.5rem;
}

.c1 > svg {
  height: 1rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 1rem;
}

<ul
  class="c0"
>
  <li
    aria-labelledby=":r0:"
    class="c1"
  >
    <span
      class="c2"
      color="currentColor"
      id=":r0:"
    >
      Feature 1
    </span>
  </li>
  <li
    aria-labelledby=":r1:"
    class="c1"
  >
    <span
      class="c2"
      color="currentColor"
      id=":r1:"
    >
      Feature 2
    </span>
  </li>
</ul>
`;
